crates/ruff_server/src/edit/replacement.rs

```diff
@@ -15,84 +16,213 @@ impl Replacement {
         modified_line_starts: &[TextSize],
     ) -> Self {
         let mut source_start = TextSize::default();
-        let mut replaced_start = TextSize::default();
-        let mut source_end = source.text_len();
-        let mut replaced_end = modified.text_len();
-        let mut line_iter = source_line_starts
+        let mut modified_start = TextSize::default();
+
+        for (source_line_start, modified_line_start) in source_line_starts
             .iter()
             .copied()
-            .zip(modified_line_starts.iter().copied());
-        for (source_line_start, modified_line_start) in line_iter.by_ref() {
-            if source_line_start != modified_line_start
-                || source[TextRange::new(source_start, source_line_start)]
-                    != modified[TextRange::new(replaced_start, modified_line_start)]
+            .zip(modified_line_starts.iter().copied())
+            .skip(1)
```

Comment:
Skipping the first line start i.e., change (2)

Reply:
Is it guaranteed that `source_line_starts[0] == TextSize::new(0)`?

Reply:
Yes:

https://github.com/astral-sh/ruff/blob/8e6f722d9bcba39a5f0ef508e4d6e454eb86f349/crates/ruff_source_file/src/line_index.rs#L28-L32